src/callstack.cr

```diff
@@ -130,16 +130,17 @@ struct CallStack
     frame = decode_frame(repeated_frame.ip)
     if frame
       offset, sname = frame
+      sname = String.new sname
       if repeated_frame.count == 0
-        LibC.printf "[%ld] %s +%ld\n", repeated_frame.ip, sname, offset
+        STDERR.printf "[%d] %s +%d\n", repeated_frame.ip.address, sname, offset
```

Comment:
These lines use `LibC.printf` instead of `printf` in the stdlib to avoid the nonblocking IO code. I'd suggest using `LibC.dprintf` with fd 2, although you'll have to bind that.